docs/readmes/proposals/p020_n7_Interface_n_network_initiated_session_modification.md

```diff
@@ -0,0 +1,165 @@
+---
+id: p020_n7_Interface_n_network_initiated_session_modification
+title: SMF to PCF Interface and Network Initiated Session Modification
+hide_title: true
+---
+# Overview
+
+*Author: @pmishra-wavelabs*\
+*Last Updated: 07/12/2021*
+
+## Background and Objective 
+This document provides an approach for the following 
+- Implementing N7 interface towards PCF through Federation Gateway
+  -- Implement 5G Services Based Interface (SBI) for Federation Gateway
+  -- Feg Relay for SM Policy control create, update, delete and update-notify
+  -- Feg Session Proxy for SM Policy control create, update, delete and 
+     update-notify 
+- Implementing Network initiated session modification
+  -- Network Initiated Session Policy association update
+  -- Network initiated session Policy association termination
+## Use Cases
+
+#### N7 Interface to PCF
+
+1. When a SM Context is created at SMF, Sessiond sends a 
+   Npcf_SMPolicyControl_Create request to PCF. The policy decision received 
+   is updated for the session context and propogated to pipelined.
+2. When a SM Context is deleted at SMF, Sessiond sends a 
+   Npcf_SMPolicyControl_delete request to the PCF.
+
+#### Network initiated Session Modification
+1. When using local policydb and when static rules are enabled or disabled, 
+   Npcf_SMPolicyControl_UpdateNotify is sent to the SessionD to install or 
+   remove the static rules.
+2. When PCF sends a Npcf_SMPolicyControl_UpdateNotify to the SessionD the 
+   following actions are performed by the SessionD
+   - Update of policies associated with the PDU session
+   - Deletion of SM Policy Association of a PDU sessions
+   - Provisioning of PCC rules
+   - Provisioning of policy control triggers
+   - Provisioning of revalidation time
+   - Policy provisioning and enforcement of authorized AMBR per PDU session
+
+## Detailed Design
+![Network initiated Session Modification and N7 Interface](../assets/proposals/p019/network_initiated_session_modification_n_n7_interface.png)
+
+#### Service Based Interface
+This component implements the SBI defined in 3GPP TS 29.500 and 29.501. 
+SBI is a HTTP/2 protocol based interface. SBI is composed of both HTTP/2 client 
+and server. The client is used to send requests and the server is used to 
+receive callbacks and notifications. This SBI is part of Feg Session Proxy
+for the N7 interface.
+
+#### Session Proxy
+Session Proxy is reused for interfacing with the PCF through N7 interface. 
+The CentralSessionController GRPC service implementation in the Session Proxy 
+invokes the N7 SBI requests when M5GSMSessionContext is received in the 
+Create/Update/Terminate session requests.
```

Comment:
grpc server is an interface in go, so no need to modify session_proxy. A new service can implement the same methods as session_proxy and receive the same messages.

Reply:
Magma is microservice architecture & session_proxy is already pretty complicated & hardly "micro". If a significant new logic/functionality is required - I'd suggest implementing it in a new service.